### PR TITLE
Support for legal NLP datasets (EURLEX, ECtHR cases and EU-REG-IR)

### PR DESCRIPTION
**Pull request by iliaschalkidis**
Add support for two legal NLP datasets:

- EURLEX (https://www.aclweb.org/anthology/P19-1636/)
- ECtHR cases (https://arxiv.org/abs/2103.13084)
- EU-REG-IR (https://arxiv.org/abs/2101.10726)

### CODE REVIEW

File: datasets/eurlex/README.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    - [Citation Information](#citation-information)
+    - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** http://nlp.cs.aueb.gr/software_and_datasets/EURLEX57K/
+- **Repository:** http://nlp.cs.aueb.gr/software_and_datasets/EURLEX57K/
+- **Paper:** https://www.aclweb.org/anthology/P19-1636/
+- **Leaderboard:** N/A
+- **Point of Contact:** Ilias Chalkidis ( ihalk[at]aueb.gr )
```
Comment (lhoestq) [inline datasets/eurlex/README.md:69]:
```suggestion
- **Point of Contact:** [Ilias Chalkidis](mailto:ihalk@aueb.gr)
```

File: datasets/eurlex/README.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+### Languages
+
+All documents are written in English.
+
+## Dataset Structure
+
+### Data Instances
+
+
+```json
```
Comment (lhoestq) [inline datasets/eurlex/README.md:103]: same suggestions as for the other dataset card:
- use an actual example from python
- crop it if it's too long

File: datasets/eurlex/README.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+The following data fields are provided for documents (`train`, `dev`, `test`):
+
+`celex_id`: The official ID of the document. The CELEX number is the unique identifier for all publications in both Eur-Lex and CELLAR.\
+`document_type`: The type of the document (e.g., Directive, Regulation, Decision)\
+`title`: The title of the document.\
+`header`: The first part of the document, which includes the title and name of the legal body enforcing the legal act.\
+`recitals`: The second part of the document, which contains legal background references.\
+`main_body`: The main part of the document, organized in articles or even bigger units (e.g., chapters, sections).\
+`eurovoc_concepts`: The relevant EUROVOC concepts (labels).
```
Comment (lhoestq) [inline datasets/eurlex/README.md:125]: same suggestions as for the other dataset card:
- remove the \
- mention the fields that are actually available in the dataset loaded in python: celex_id, title, text, eurovoc_concepts
- mention the type of each field

File: datasets/eurlex/README.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+For simplicity, we provide the full content (`text`) of each document, which is represented by its `header`, `recitals` and `main_body`.
+
+If you want to use the descriptors of EUROVOC concepts, similar to Chalkidis et al. (2020), please load: https://archive.org/download/EURLEX57K/eurovoc_concepts.jsonl
```
Comment (lhoestq) [inline datasets/eurlex/README.md:129]: They are already included in the dataset loaded with `load_dataset` no ?

File: datasets/eurlex/README.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+N/A
+
+### Discussion of Biases
+
+N/A
+
+### Other Known Limitations
+
+N/A
```
Comment (lhoestq) [inline datasets/eurlex/README.md:192]:
```suggestion
[More Information Needed]

### Discussion of Biases

[More Information Needed]

### Other Known Limitations

[More Information Needed]
```

**Review by lhoestq — COMMENTED** [review]
Awesome thank you :)
This is really cool

I left a few comments.

Also it looks like the dummy data are quite big (100-200KB each). Can you try to reduce their sizes please ? For example I noticed that all the jsonl files inside the `dummy_data.zip` files have 20 lines. Can you only keep 2 lines instead ?

File: datasets/eurlex/README.md (hunk 4 above)
Comment (iliaschalkidis) [reply datasets/eurlex/README.md:121]: I updated README to represent the actual objects.

File: datasets/eurlex/README.md (hunk 2 above)
Comment (iliaschalkidis) [reply datasets/eurlex/README.md:103]: I used a actual example as loaded with `load_dataset`

**Comment by iliaschalkidis** [comment]
> Awesome thank you :)
> This is really cool
> 
> I left a few comments.
> 
> Also it looks like the dummy data are quite big (100-200KB each). Can you try to reduce their sizes please ? For example I noticed that all the jsonl files inside the `dummy_data.zip` files have 20 lines. Can you only keep 2 lines instead ?

Hi @lhoestq, I did my best to improve the README files, while I also decreased dummy data examples.

File: datasets/ecthr_cases/ecthr_cases.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    "P7-5": "Equality between spouses",
+    "P12-1": "General prohibition of discrimination",
+    "P13-1": "Abolition of the death penalty",
+    "P13-2": "Prohibition of derogations",
+    "P13-3": "Prohibition of reservations",
+}
+
+
+# TODO: Name of the dataset usually match the script name with CamelCase instead of snake_case
+class ECtHRCases(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/ecthr_cases/ecthr_cases.py:95]:
```suggestion
class EcthrCases(datasets.GeneratorBasedBuilder):
```
This must be the camel case version of the dataset script name `etchr_cases`

File: datasets/eu_regulatory_ir/README.md [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+    - [Citation Information](#citation-information)
+    - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** https://archive.org/details/eacl2021_regir_datasets
+- **Repository:** https://archive.org/details/eacl2021_regir_datasets
+- **Paper:** https://arxiv.org/abs/2101.10726
+- **Leaderboard:** N/A
+- **Point of Contact:** Ilias Chalkidis ( ihalk[at]aueb.gr )
```
Comment (lhoestq) [inline datasets/eu_regulatory_ir/README.md:69]:
```suggestion
- **Point of Contact:** [Ilias Chalkidis](mailto:ihalk@aueb.gr)
```

File: datasets/eu_regulatory_ir/README.md [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+### Languages
+
+All documents are written in English.
+
+## Dataset Structure
+
+### Data Instances
+
+```json
+{
```
Comment (lhoestq) [inline datasets/eu_regulatory_ir/README.md:97]: this example is too long, can you crop it ?

**Review by lhoestq — COMMENTED** [review]
Nice !
And thanks for adding this 3rd legal dataset :)

I added my final comments :)
Good job !

File: datasets/eu_regulatory_ir/eu_regulatory_ir.py [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    """EURegIR: Regulatory Compliance IR (EU/UK)"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="eu2uk", version=VERSION, description="EURegIR: Regulatory Compliance IR (EU2UK)"),
+        datasets.BuilderConfig(name="uk2eu", version=VERSION, description="EURegIR: Regulatory Compliance IR (UK2EU)"),
+    ]
+
+    DEFAULT_CONFIG_NAME = "eu2uk"  # It's not mandatory to have a default configuration. Just use one if it make sense.
```
[the remark this answers is not in the document]
Comment (iliaschalkidis) [reply datasets/eu_regulatory_ir/eu_regulatory_ir.py]: Removed.

File: datasets/eu_regulatory_ir/README.md (hunk 8 above)
Comment (iliaschalkidis) [reply datasets/eu_regulatory_ir/README.md:89]: Done

**Comment by iliaschalkidis** [comment]
@lhoestq thanks for your review.

 I shortened the examples in README files and removed `DEFAULT_CONFIG_BUILDER` from `eu_regulatory_ir.py`.

**Pull request merged**